bridge-impl/src/main/java/com/liferay/faces/bridge/renderkit/html_basic/internal/ResponseWriterResourceImpl.java

```diff
@@ -30,74 +31,273 @@
 /**
  * The purpose of this class is to bypass the Mojarra/MyFaces encoding of the URL found in the "src" attribute of a
  * &lt;script&gt; element or the "href" attribute of a &lt;link&gt; element. For more info, see: <a
- * href="http://issues.liferay.com/browse/FACES-1236">FACES-1236</a>.
+ * href="http://issues.liferay.com/browse/FACES-1236">FACES-1236</a> and <a
+ * href="https://github.com/javaserverfaces/mojarra/issues/4345">MOJARRA #4345</a>. This class also ensures that body
+ * script resources are loaded in each Ajax request by rendering <code>type="text/javascript"</code> as the last
+ * attribute. For more info, see: <a href="https://issues.liferay.com/browse/FACES-3252">FACES-3252</a> and <a
+ * href="https://github.com/javaserverfaces/mojarra/issues/4340">MOJARRA #4340</a>.
  *
  * @author  Neil Griffin
  */
 public class ResponseWriterResourceImpl extends ResponseWriterWrapper {
 
-	// Private Constants
-	private static final String REGEX_AMPERSAND = "[&]amp;";
+	// Private Final Data Members
+	private final FacesURLEncoder facesURLEncoder;
+	private final ResponseWriter wrappedResponseWriter;
 
 	// Private Data Members
-	FacesURLEncoder facesURLEncoder;
-	private ResponseWriter wrappedResponseWriter;
+	private Attribute javaScriptType;
+	private boolean writingLink;
+	private boolean writingScript;
 
 	public ResponseWriterResourceImpl(FacesContext facesContext, ResponseWriter responseWriter) {
 		ExternalContext externalContext = facesContext.getExternalContext();
 		this.facesURLEncoder = FacesURLEncoderFactory.getFacesURLEncoderInstance(externalContext);
 		this.wrappedResponseWriter = responseWriter;
 	}
 
+	@Override
+	public Writer append(char c) throws IOException {
+
+		writeTypeTextJavaScriptIfNecessary();
```

Comment:
@stiemannkj1 I see why it is necessary to call `writeTypeTextJavaScriptIfNecessary()` from methods like `close()` and `endDocument()` but not sure how it helps to call it from the `append(...)` methods?

Reply:
@stiemannkj1 Never-mind, I think I get it. Methods like `append` and `write` are writing in the <script>inside-of-the-script-element</script> so that makes sense.

Reply:
I assumed that the `append()` methods behaved in the same way as their `write()` counterparts but just returned the writer. I guess they don't after looking at `HtmlResponseWriter` in Mojarra. We can keep that code in or remove it. It's up to you. Also it ooks like `ResponseWriter.append()` is only called in `DoctypeRenderer` and `RenderKitUtils` in Mojarra. I don't know if that matters or not.